company/adobe/AddDigits.java

```diff
@@ -5,16 +5,17 @@
 
 //Follow up:
 //Could you do it without any loop/recursion in O(1) runtime?
+// I am able to solve without using the extra while loop.
 
 class AddDigits {
     public int addDigits(int num) {
-        while(num >= 10) {
-            int temp = 0;
-            while(num > 0) {
-                temp += num % 10;
-                num /= 10;
-            }
-            num = temp;
+        int x = 0, y = 0;
+        int sum = 0;
+        while(num >=10) {
+            x = num % 10;
+            y = num / 10;
+            sum = x+y;
```

Comment:
i think this only happen in interview： 
num = 2^32  (sum = x + y).

Reply:
cool, i thought while loop will contribute n*n complexity. What I mean is-

so in the above case  while(num >=10) {} while loop is running until the number is greater than or equal to 10, 
 so it would run and have the complexity like num*num*num until the condition is satisfied instead of 2^32?
Hope I am making some sense?
Thanks